pkg/util/util.go

```diff
@@ -79,10 +79,11 @@ func GetLocalCluster(ss types.SubmarinerSpecification) (types.SubmarinerCluster,
 	localCluster.Spec.ColorCodes = ss.ColorCodes
 	return localCluster, nil
 }
+
 func GetLocalEndpoint(clusterID string, backend string, backendConfig map[string]string, natEnabled bool, subnets []string) (types.SubmarinerEndpoint, error) {
 	hostname, err := os.Hostname()
 	if err != nil {
-		klog.Fatalf("error getting hostname: %s", err.Error())
+		return types.SubmarinerEndpoint{}, fmt.Errorf("Error getting hostname: %v", err)
```

Comment:
Another option could be returning a pointer to SubmarinerEndpoint, that let's you return nil in case of error.

Reply:
yeah that crossed my mind but would require changes elsewhere as well. I can look into that in a separate patch.